drv/system.go

```diff
@@ -1,19 +1,148 @@
 package drv
 
+import (
+	"encoding/json"
+	"os/exec"
+	"strings"
+	"time"
+)
+
+type bootcStatus struct {
+	Status struct {
+		Booted struct {
+			Incompatible bool `json:"incompatible"`
+			Image        struct {
+				Timestamp string `json:"timestamp"`
+			} `json:"image"`
+		} `json:"booted"`
+		Staged struct {
+			Incompatible bool `json:"incompatible"`
+			Image        struct {
+				Timestamp string `json:"timestamp"`
+			} `json:"image"`
+		}
+	} `json:"status"`
+}
+
+// TODO: Temporary: WILL get removed at some point.
+type rpmOstreeStatus struct {
+	Deployments []struct {
+		Timestamp int64 `json:"timestamp"`
+	} `json:"deployments"`
+}
+
 type SystemUpdater struct {
 	Config          DriverConfiguration
-	SystemDriver    SystemUpdateDriver
+	SystemDriver    SystemDriver
```

Comment:
I think having seperate update drivers for rpm-ostree and bootc would be better than having a case statement in every function. The plan is to eventually deprecate rpm-ostree so it would be easier to just remove it here entirely. It would likely mean there are more LoC, but it would make more sense semantically, as they are technically different "drivers"

Reply:
This would be similar to before except likely a bit more organized

Reply:
Do you mind if I do this change on a new PR? The ones dependent on this one change it a bit because of the env changes. It should be pretty easy to split them afterwards

Reply:
Worst case I can just change it here and fix the merge conflicts

Reply:
> Do you mind if I do this change on a new PR? The ones dependent on this one change it a bit because of the env changes. It should be pretty easy to split them afterwards

I don't mind, feel free to make a new PR

Reply:
Ill mark this one as a draft until then. Ill check out the other reviews and fix them first